Source/raceQuestPawn/patch_PawnGenerator_GeneratePawn.cs

```diff
@@ -70,7 +71,7 @@ public static void Prefix(ref PawnGenerationRequest request)
                         {
                             if (!allPawnKinds.Contains(pawnGenOption.kind))
                             {
-                                allPawnKinds.Add(pawnGenOption.kind);
+                                allPawnKinds.Prepend(pawnGenOption.kind);
```

Comment:
Not sure this is correct, Prepend returns the changed list but the return-valuie is never used. So the list is empty and nothing is done.

Reply:
Anyway, this does work in my save, since I guess the pawnkind of "main pawn" in caravan is given a "xxx trader" kind but the this mod use an opposite sequence against the original sequence of the pawnkind distributor, so the "main pawn" with "xxx trader" kind get another pawnkind, then comes the bug.

Reply:
@emipa606 Are you test with debug menu "add guest"? It's broken in vanilla game, doesn't do anything

Reply:
I've figure out another and the most possible solution, wait some time